### ISSUE TITLE
How to use only one component without applying the whole theme styles to the application?

### ISSUE THREAD

**Issue by henryyue2010** [akveo/nebular#1663]
I am trying to use NbChatComponent in my Angular application without applying the whole Nebular theme styles. How to get the default nebular style only apply to the Angular component which uses NbChatComponent? Any samples?
Thank you very much in advance for your help.
Henry

**Comment by yggg** [comment]
Unfortunately, there is no out of the box solution.

You can do this by calling mixins for individual components inside `nb-install`  but this not a super solid solution though. Since chat is using a button, input and icon components under the hood you need to include styles for these components as well. Also, you need to keep track of module dependencies added in the future as it potentially could require more styles to be added.

To sum up:
1. Setup [custom theming](https://akveo.github.io/nebular/docs/design-system/enable-customizable-theme#enable-customizable-themes). You can skip modifying theme variables
2. Change `styles.scss` to have styles in `nb-install` only for the chat and it's dependencies:
```
@include nb-install() {
  @include nb-chat-theme();
  @include nb-icon-theme();
  @include nb-buttons-theme();
  @include nb-input-theme();
};
```

**Comment by filol** [comment]
Hi @yggg , I'm trying to do exactly the same thing, ie import the component chat only into a component. I use for the rest of my Material Angular application and it seems that the 2 comes into conflict (it's a shame). I did what you said but I'm still full of error type `ERROR TypeError: Can not read property 'appendChild' of undefined`, distorted images, ... How to really limit nebular to a single component?

**Comment by AmmarIsmaeel** [comment]
Me too, I am having the same problem and I really need and answer How to really limit nebular to a single component?

**Comment by el7or** [comment]
Me too, I am having the same problem and I really need and answer How to import the component chat only into my application?

**Comment by dinohorvat** [comment]
You can use Angular Elements (Web components)

**Comment by KagariSan** [comment]
> Me too, I am having the same problem and I really need and answer How to import the chat component only into my application without applying the whole theme styles to the application?

Hi, there is how I use only one component for project without apply style to another component.

- First: install it normally by running `ng add @nebular/theme ` or by manually.
- Second: if you install it by running command then go to AppComponent, revert what nebular added to your template like this one
`
 <nb-layout>
      <nb-layout-header fixed>Company Name</nb-layout-header>

      <nb-sidebar>Sidebar Content</nb-sidebar>

      <nb-layout-column>
        Page Content <button nbButton>Hello World</button>
      </nb-layout-column>
    </nb-layout>
`
- After this, you can use nebular component by grab it between
`
 <nb-layout>
      <nb-layout-column>
 <Your nebular HTML go here>
      </nb-layout-column>
    </nb-layout>
`

P/S: there will be redundant over there and there but at least nebular style will not apply over other component of your project. Maybe better solution out there, better keep looking :eyes:

**Comment by Alec2435** [comment]
For those dealing with the problem Nebular causes with `ERROR TypeError: Can not read property 'appendChild' of undefined` while trying to use just a single Nebular component, there's an easy fix I just found.

Go to your `app.module.ts` and in the `providers` array (add one if you don't have one) add `{ provide: OverlayContainer }` and it should fix the issue for the most part.

**Comment by truonghoangduy** [comment]
> For those dealing with the problem Nebular causes with `ERROR TypeError: Can not read property 'appendChild' of undefined` while trying to use just a single Nebular component, there's an easy fix I just found.
> 
> Go to your `app.module.ts` and in the `providers` array (add one if you don't have one) add `{ provide: OverlayContainer }` and it should fix the issue for the most part.

You save my date 👍

My Angular Tree
<nb-layout>
<nb-side-nav></nb-side-nav>
      <nb-layout-column>
            <nb-layout>
                  <nb-side-nav></nb-side-nav>
             </nb-layout>
       </nb-layout-column>
</nb-layout>

**Comment by tegola** [comment]
I have managed to solve this. See [here](https://github.com/akveo/nebular/issues/2703#issuecomment-856668538).

**Comment by eliasmatheus** [comment]
> For those dealing with the problem Nebular causes with `ERROR TypeError: Can not read property 'appendChild' of undefined` while trying to use just a single Nebular component, there's an easy fix I just found.
> 
> Go to your `app.module.ts` and in the `providers` array (add one if you don't have one) add `{ provide: OverlayContainer }` and it should fix the issue for the most part.

Wow! That worked. 

Thank you!

**Comment by CleydermanG** [comment]
> > For those dealing with the problem Nebular causes with `ERROR TypeError: Can not read property 'appendChild' of undefined` while trying to use just a single Nebular component, there's an easy fix I just found.
> > Go to your `app.module.ts` and in the `providers` array (add one if you don't have one) add `{ provide: OverlayContainer }` and it should fix the issue for the most part.
> 
> You save my date 👍
> 
> My Angular Tree

Thank you!!

**Comment by CleydermanG** [comment]
Angular

**Comment by AhmedMansour92** [comment]
Hi @henryyue2010
### Solved

### 1.  add this line in imports of `app.module.ts`
```
imports: [  
    NbThemeModule.forRoot({ name: 'default' }),
]
```

### 2.`app.component.html` ;
```
<nb-layout>
	<nb-layout-column>
		<main>
			<router-outlet></router-outlet>
		</main>
	</nb-layout-column>
</nb-layout>

```

### 3. Make sure that the following code is part of `styles.scss` ;
```
@use 'themes' as *;
@use '@nebular/theme/styles/globals' as *;
@include nb-install() {
  @include nb-theme-global();
};

```

### 4. `themes.scss` ;
```
@forward '@nebular/theme/styles/theming';
@use '@nebular/theme/styles/theming' as *;
@use '@nebular/theme/styles/themes/default';

$nb-themes: nb-register-theme((
  // add your variables here like:
  // color-primary-100: #f2f6ff,
  // color-primary-200: #d9e4ff,
), default, default);

```

### 5. in  `app.component.ts` add this piece of code in `ngOnInit()`
```
  ngOnInit(): void {
      this.router.events.subscribe((event) => {
          document.body.classList.remove('nb-theme-default');
      });
  }
```

- _This will remove the nebular theme from all the app._


### 6. in the component you want to apply the theme add this in `ngOnInit()`
```
  ngOnInit(): void {
    document.body.classList.add('nb-theme-default');
  }
```

-  _This will add the nebular theme @ the current component._

**Comment by jonkenobi** [comment]
I solved it by doing [this](https://github.com/akveo/nebular/issues/2461#issuecomment-1225232749)
